substrate/frame/child-bounties/src/migration.rs

```diff
@@ -0,0 +1,229 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+use core::marker::PhantomData;
+use frame_support::{
+	storage_alias,
+	traits::{Get, UncheckedOnRuntimeUpgrade},
+};
+
+use alloc::collections::BTreeSet;
+#[cfg(feature = "try-runtime")]
+use alloc::vec::Vec;
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+
+pub mod v1 {
+	use super::*;
+
+	/// Creates a new ids for the child balances based on the child bounty count per parent bounty
+	/// instead of the total child bounty count. Translates the existing child bounties to the new
+	/// ids. Creates the `V0ToV1ChildBountyIds` map from `old_child_id` to new (`parent_id`,
+	/// `new_child_id`).
+	///
+	/// `TransferWeight` returns `Weight` of `T::Currency::transfer` and `T::Currency::free_balance`
+	/// operation which is performed during this migration.
+	pub struct MigrateToV1Impl<T, TransferWeight>(PhantomData<(T, TransferWeight)>);
+
+	#[storage_alias]
+	type ChildBountyDescriptions<T: Config + pallet_bounties::Config> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		BountyIndex,
+		BoundedVec<u8, <T as pallet_bounties::Config>::MaximumReasonLength>,
```

Comment:
I think it should be fine to use a`WeakBoundedVec` here, just in case when someone decreases `MaximumReasonLength` at the same time as applying the migration. Their encodings should be identical.